Modules/Core/SpatialObjects/include/itkSpatialObjectToImageStatisticsCalculator.hxx

```diff
@@ -151,7 +151,7 @@ SpatialObjectToImageStatisticsCalculator< TInputImage, TInputSpatialObject,
         maskImage->TransformIndexToPhysicalPoint( ind, pnt );
         tPnt = maskSpatialObject->GetObjectToWorldTransform()
           ->TransformPoint( pnt );
-        m_Image->TransformPhysicalPointToIndex( tPnt, ind );
+        ind = m_Image->TransformPhysicalPointToIndex(tPnt);
```

Comment:
For cases like this without a new left hand side variable, do you think an extra copy operation is being introduced?

Reply:
Thanks for your question @blowekamp I don't think returning an `itk::Index` object and copying this object into a variable (`ind`) is in any way more expensive than modifying the `ind` inside the function via pass-by-reference.  `itk::Index` objects have cheap, fast copy-semantics (no dynamic memory allocation involved).

Reply:
I am quite curious if there is a performance difference. My intution ( experience)  is that doing it in-place would be slightly better unless there is something similar to C++ RVO. It would be quite good to have concrete evidence of the performance difference. 

Would you like to do the profiling?

Reply:
I just observed that the new (ITK5) `TransformPhysicalPointToIndex(point)` overload is more than 20% _faster_ than the original  `TransformPhysicalPointToIndex(point, index)`. :smile:

I also looked specifically at `SpatialObjectToImageStatisticsCalculator::Update()`. But honestly I could not observe a significant performance effect of this commit on `calculator->Update()`. Probably because the `Update()` does so many other things than just `TransformPhysicalPointToIndex`.

Reply:
Nice work! It is always good to do this type of analysis to determine "best practices" and recommendations as we continue to move forward. Here we can say that the new method is recommended when the `isInBounds` is not needed.